examples/tutorial/lessons/lesson-04-lifecycle.adoc

```diff
@@ -0,0 +1,366 @@
+  "text": "I did it!"
+  }
+],
+"lifecycle": [
+  {
+  "status": "OPEN",
+  "user": "system",
+  "stime": 1488484795014
+  },
+  {
+  "status": "ACKNOWLEDGED",
+  "user": "RacerX",
+  "stime": 148854994066
+  },
+  {
+  "status": "RESOLVED",
+  "user": "racerX",
+  "stime": 1488551286972
+  }
+]
+----
+
+Now that our alert is resolved, let's check back on the lesson-03 trigger.  Fetch the lesson-03 trigger (the request URL is above).  You should see that the trigger has been auto-enabled and now shows `"enabled": true`.
+
+=== Auto-Resolve
+
+Hawkular Alerting provides a powerful feature called _Auto-Resolve_.  We have already learned about _FIRING_ conditions, the condition set that defines when we have detected an anomaly and a trigger can fire.  We can also define _AUTORESOLVE_ conditions, the condition set that defines when the anomaly has been resolved.  Triggers with `autoResolve=true` alternate between FIRING mode and AUTORESOLVE mode.  As long as an issue is fixed and a system returns to acceptable behavior, it is not necessary to worry about manually updating alert life-cycle.
+
+Note that when `autoResolve=true` the `autoDisable` and `autoEnable` options are *ignored*. An auto-resolve trigger toggles between FIRING mode and AUTORESOLVE mode so by its nature it autoDisables and autoEnables.
+
+The AUTORESOLVE condition set can logically be thought of as the opposite of the FIRING condition set. But in practice it is often not just the negation of the FIRING condition set.  Also, the AUTORESOLVE condition set can define its own dampening.  For example, an alert may be raised due to a shorter spike of high response times but we may not want to declare the issue resolved until we have a longer period of acceptable response times. http://www.hawkular.org/community/docs/developer-guide/alerts.html#_autoresolve[See here for more detail] on the auto resolve feature.  Now, let's try an example to see it in action.
+
+First, let's delete the lesson-03 trigger, so we don't get any more lesson-03 alerts.
+
+[cols="1,5l"]
+|===
+|Request
+|DELETE http://localhost:8080/hawkular/alerts/triggers/lesson-03
+
+|HTTP Headers
+|Hawkular-Tenant: tutorial
+Content-Type: application/json
+|===
+
+Now, let's define a lesson-04 trigger that fires if the gauge-1 metric falls outside of the range [20..50], and resolves automatically when gauge-1 stays within the desired range of [25..45], at least 2 times out of 3 evaluations. We'll use a full trigger definition:
+
+[cols="1,5l"]
+|===
+|Request
+|POST http://localhost:8080/hawkular/alerts/triggers/trigger
+
+|HTTP Headers
+|Hawkular-Tenant: tutorial
+Content-Type: application/json
+
+|Body
+|
+{
+"trigger": {
+  "tenantId": "tutorial",
+  "id": "lesson-04",
+  "name": "Lesson 03",
+  "description": "An Auto-Resolving Trigger",
+  "autoResolve": true,
+  "autoResolveAlerts": true,
+  "enabled": true,
+  "firingMatch": "ALL",
+  "autoResolveMatch": "ALL"
+},
+"conditions": [
+  {
+    "tenantId": "tutorial",
```

Comment:
Same comment, using full triggers no need to repeat tenantId/triggerId on dampenings and conditions.